GooglePayment/src/main/java/com/braintreepayments/api/GooglePaymentClient.java

```diff
@@ -166,133 +162,133 @@ public void onComplete(@NonNull Task<Boolean> task) {
      * @param request  The {@link GooglePaymentRequest} containing options for the transaction.
+     * @param callback Instance of {@link GooglePaymentRequestPaymentCallback} to receive the result.
      */
-    public static void requestPayment(final BraintreeFragment fragment, final @NonNull GooglePaymentRequest request) {
-        fragment.sendAnalyticsEvent("google-payment.selected");
+    public void requestPayment(final FragmentActivity activity, final GooglePaymentRequest request, final GooglePaymentRequestPaymentCallback callback) {
+        braintreeClient.sendAnalyticsEvent(activity, "google-payment.selected");
 
-        if (!validateManifest(fragment.getApplicationContext())) {
-            fragment.postCallback(new BraintreeException("GooglePaymentActivity was not found in the Android " +
+        if (!validateManifest(activity)) {
+            callback.onResult(false, new BraintreeException("GooglePaymentActivity was not found in the Android " +
                     "manifest, or did not have a theme of R.style.bt_transparent_activity"));
-            fragment.sendAnalyticsEvent("google-payment.failed");
+            braintreeClient.sendAnalyticsEvent(activity,"google-payment.failed");
             return;
         }
 
         if (request == null) {
-            fragment.postCallback(new BraintreeException("Cannot pass null GooglePaymentRequest to requestPayment"));
-            fragment.sendAnalyticsEvent("google-payment.failed");
+            callback.onResult(false, new BraintreeException("Cannot pass null GooglePaymentRequest to requestPayment"));
+            braintreeClient.sendAnalyticsEvent(activity, "google-payment.failed");
             return;
         }
 
         if (request.getTransactionInfo() == null) {
-            fragment.postCallback(new BraintreeException("Cannot pass null TransactionInfo to requestPayment"));
-            fragment.sendAnalyticsEvent("google-payment.failed");
+            callback.onResult(false, new BraintreeException("Cannot pass null TransactionInfo to requestPayment"));
+            braintreeClient.sendAnalyticsEvent(activity, "google-payment.failed");
             return;
         }
 
-        fragment.waitForConfiguration(new ConfigurationListener() {
+        braintreeClient.getConfiguration(activity, new ConfigurationCallback() {
             @Override
-            public void onConfigurationFetched(Configuration configuration) {
-
-                if (!configuration.getGooglePayment().isEnabled(fragment.getApplicationContext())) {
-                    fragment.postCallback(new BraintreeException("Google Pay enabled is not enabled for your Braintree account," +
+            public void onResult(@Nullable Configuration configuration, @Nullable Exception e) {
+                if(!configuration.getGooglePayment().isEnabled()) {
+                    callback.onResult(false, new BraintreeException("Google Pay enabled is not enabled for your Braintree account," +
```

Comment:
This message is a little awkward, the first instance of `enabled` can probably be removed.

Reply:
Yeah def. We'll update.